cirq/api/google/v1/hardware.proto

```diff
@@ -0,0 +1,114 @@
+syntax = "proto3";
+
+import "cirq/api/google/v1/operations.proto";
+import "cirq/api/google/v1/params.proto";
+
+package cirq.api.google.v1;
+
+option java_package = "com.google.cirq.api.google.v1";
+option java_outer_classname = "HardwareProto";
+option java_multiple_files = true;
+
+// The operations supported by the hardware and the current performance
+// characteristics of these operations.
+message HardwareSpec {
+  // The version name of the hardware.
+  //
+  // Currently supports: `bristlecone`.
+  string processor_name = 1;
+
+  // The supported operations on the hardware. This information is static
+  // and does not change for a given processor_name.
+  GateSet gate_set = 2;
+
+  // The expected performance characteristics of the hardware at a particular
+  // snapshot in time.
+  Performance performance = 3;
+}
+
+// Gate prototypes, representing a gate that can be applied to specific qubits.
+message Prototype {
+  // A prototype is a gate plus the qubits upon which the gate acts.
+  //
+  // If the gate supports parameterized gates, then the parameterized floats
+  // will have string keys that can be mapped to accuracies and float ranges
+  // (see `parameter_ranges` in `GateSet`.)
+  //
+  // There is a prototype for every qubits upon which the gate can act.
+  //
+  // Some of the prototype operation fields will not be set. These include
+  // the `key` and `invert_mask` on measurement gates, and
+  // `incremental_delay_picoseconds` on all operations
+  Operation operation = 1;
+
+  // The resolution with which the gate can be offset (values of
+  // `incremental_delay_picoseconds` will be rounded to values in increments
+  // of this size).
+  uint64 timing_resolution_picoseconds = 2;
+}
+
+// The range that a parameter can take along with the accuracy with which
+// the parameter can be supplied.
+message ParameterRange {
+  // The lowest allowed value for the parameter.
```

Comment:
Will parameters always be linear? Maybe we should switch from `step_size` to `distinct_value_count` and add an enum `interpolation` set to `linear`.